table/block_based/block_based_table_reader.cc

```diff
@@ -637,20 +637,34 @@ Status BlockBasedTable::Open(
     auto version_iter = uprops.find(ExternalSstFilePropertyNames::kVersion);
     if (version_iter == uprops.end()) {
       // Normal (non-external) SST file - can only use embedded db_session_id
-      // with current file number (which should be original file number)
-      if (file_num > 0) {
+      // if original or current file number are known (which should be same)
+      if (rep->table_properties->orig_file_number > 0) {
```

Comment:
Is there any case that the original file number of an SST file is different from the current_file_number?

Reply:
For SST files not generated by SstFileWriter, I'm not aware of any RocksDB infrastructure that changes the file number or file name (after last '/') after it is initially written. I'm pushing a change with an assertion here to check for this.

But we have to deal with cases when orig_file_number is not known (old SST file) or current file number is not known (depending on how the file reader is initialized).

Reply:
It looks like import_column_family_test shows that my assumption was wrong, and we cannot rely on current file number for stable cache key in any case. Regardless of normal vs. external SST file, we either have to use <orig_db_session_id,orig_file_number> entirely from table properties if available, or fall back on <cur_db_session_id,cur_file_number>. At least that simplifies the logic.